ablySpec/RealtimeClientChannel.swift

```diff
@@ -13,6 +13,83 @@ class RealtimeClientChannel: QuickSpec {
     override func spec() {
         describe("Channel") {
 
+            // RTL3
+            context("connection state") {
+
+                // RTL3b
+                context("changes to SUSPENDED") {
+
+                    it("ATTACHING channel should transition to DETACHED") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        let transport = client.transport as! TestProxyTransport
+                        transport.actionsIgnored += [.Attached]
+
+                        expect(client.connection().state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Attaching))
+                        client.onSuspended()
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
+                    }
+
+                    it("ATTACHED channel should transition to DETACHED") {
+                        let options = AblyTests.commonAppSetup()
+                        let client = ARTRealtime(options: options)
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Attached), timeout: testTimeout)
+                        client.onSuspended()
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
+                    }
+
+                }
+
+                // RTL3b
+                context("changes to CLOSED") {
+
+                    it("ATTACHING channel should transition to DETACHED") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channel = client.channel("test")
+                        channel.attach()
+                        let transport = client.transport as! TestProxyTransport
+                        transport.actionsIgnored += [.Attached]
+
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Attaching))
+                        client.close()
+                        expect(channel.state).to(equal(ARTRealtimeChannelState.Detached))
```

Comment:
Shouldn't this be eventually as the state after a `close` would be `closing` and the channel state would still be attached at this point?